### ISSUE TITLE
Handle for merging or moving a terminal window

### ISSUE THREAD

**Issue by srv-code** [microsoft/terminal#4133]
Please provide a handle to:
1. Move a window opened in either vertical or horizontal split to a new window as a tab.
2. Move a tab inside another tab in either vertical or horizontal split (depending upon the dropping region, bottom region suggests to split horizontally and side-wise dropping suggests vertical split similar to MS Visual Studio window re-arrangement mechanism).
3. Interchange horizontal and vertical split modes.
4. Move a window in either of the split modes or tab to a new app window (in another app instance).

**Comment by zadjii-msft** [comment]
As I mentioned in #3386:
> Please don't file giant issues with multiple issues reported like this one. It makes triaging and tracking the individual component issues much harder.

1. Is this "drag a pane out to a new window"? That's probably not possible with our current architecture, but something that could be possible after #1256
2. This one's being tracked as a part of #1000, but doesn't have it's own issue, it's only tracked as this https://github.com/microsoft/terminal/issues/443#issuecomment-519077162
3. This doesn't have it's own issue yet, a new issue should be filed to discuss this feature.
4. Yea this definitely falls under the bucket of #1256.

**Comment by srv-code** [comment]
Yes. Its [#1256](https://github.com/microsoft/terminal/issues/1256).
